### ISSUE TITLE
Work on Streams that need to be done before we can start with akka-http

### ISSUE THREAD

**Issue by Silv3rcircl3** [akkadotnet/akka.net#2709]
See also #2600

- [Move compression tests over from akka-http + fix for ByteStringParser](https://github.com/akka/akka/pull/21872)
- [Move (de)compression helpers to akka-stream module ](https://github.com/akka/akka/pull/21409)
- [Fix ByteStringParser backpressure](https://github.com/akka/akka/pull/21853)
- [move compression tests](https://github.com/akka/akka/pull/21876)

**Comment by Horusiath** [comment]
Wow, are we going to port akka-http? :P This is seriously heavy task

**Comment by alexvaluyskiy** [comment]
I like Akka.Http low-level API, but High-level API seems more complicated to me

**Comment by Horusiath** [comment]
Yep, we could probably use http-core API as is, but I think that high-level (user facing) API should be something more familiar in spirit to .NET community.
